src/main/java/com/meilisearch/sdk/json/GsonJsonHandler.java

```diff
@@ -16,25 +19,27 @@ public GsonJsonHandler(Gson gson) {
 
     @Override
     public String encode(Object o) throws Exception {
-        if (o.getClass() == String.class) {
+        if (o != null && o.getClass() == String.class) {
             return (String) o;
         }
+        if (o != null && o.getClass() == Key.class) {
+            GsonBuilder builder = new GsonBuilder();
+            this.gson = builder.serializeNulls().setDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'").create();
+        }
         try {
             return gson.toJson(o);
         } catch (Exception e) {
-            // todo: use dedicated exception
-            throw new RuntimeException("Error while serializing: ", e);
+            throw new MeiliSearchRuntimeException(e);
```

Comment:
I think we can create a new custom class of errors to describe the decoding exception here :)